### ISSUE TITLE
Open dfnPanel using keyboard

### ISSUE THREAD

**Issue by sidvishnoi** [w3c/respec#2825]
Presently, [core/dfn-panel](https://github.com/w3c/respec/blob/58b9c4442d3dcc7d671a3270341d7f0da8fe4f25/src/core/dfn-panel.js) can be opened only with a mouse click. It would be good for accessibility if we can open it using keyboard. For that, `<dfn>` would need to be keyboard focusable first, and we should probably open panel on `Enter` - opening just on focus might be annoying.

**Comment by jnurthen** [comment]
I'm going to have to revert the changes I'm making to terms unless this gets resolved. I'm trying to move ARIA to the latest Respec versions but can't do so with accessibility issues in the specs

**Comment by jnurthen** [comment]
- focusable element needs an actionable role. You can't just add tabindex and a click handler - it would need to be a button or a link (my preference would be link)
- It needs to be investigated how a keyboard user would close this overlay
- The overlay which appears is currently at the end of the document. Ideally this would be moved inline In the correct place as this would simplify focus management. if this can't be done then we need to ensure that screen reader users can get focus back to the opener link easily.

**Comment by marcoscaceres** [comment]
> focusable element needs an actionable role. You can't just add tabindex and a click handler - it would need to be a button or a link (my preference would be link)

Could we assign `role="button"` to the `dfn` or some other role? Or is there a better pattern, given the aside is kinda like a little menu?

> It needs to be investigated how a keyboard user would close this overlay

Maybe leaving the corresponding `<aside>` is enough, by tabbing out of it.

> The overlay which appears is currently at the end of the document. Ideally this would be moved inline In the correct place as this would simplify focus management. 

This might be complicated, because `<dfn>s` are mostly in `<p>`s and `<li>`s. 

> if this can't be done then we need to ensure that screen reader users can get focus back to the opener link easily.

That should be doable on leaving. We know where the event started, so it should be easy to get back to where it came from.
